pyfr/solvers/base/inters.py

```diff
@@ -68,9 +68,10 @@ def _get_perm_for_view(self, inter, meth):
 
         return np.argsort(mm[0])
 
-    def _view(self, inter, meth, vshape=()):
+    def _view(self, inter, meth, vshape=(), with_perm=True):
```

Comment:
What is the with_perm needed for?

Reply:
Or, put another way, how come the constant matrix routines can get away without it?

Reply:
The views for one point per face don't use a perm

Reply:
I'm guessing they don't use any constant data either?  Maybe a generic solution is to say `self._perm if len(inter) == self.ninterfpts else Ellipsis` and then it does the right thing automatically?  Less error prone.

Reply:
I think `len(inter)` is always equal to `self.ninters`, not `self.ninterfpts`, and it's that way regardless of whether we need a perm or not (I assume since we repeat when we call entmin_int_for_inter)

Reply:
Post-concat the length should be correct as this is what does the unraveling.

Reply:
Updated

Reply:
Should I just revert to this method and explicitly tag views that need or don't need perms? Otherwise checking it by length fails for p=0

Reply:
I think the cleanest solution is to revert back to an explicit argument of some sort.